pkg/controller/statusmanager/status_manager.go

```diff
@@ -97,6 +95,11 @@ func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.Cluste
 				log.Printf("Object Kind is Namespace, skip")
 				continue
 			}
+			// @aconstan: remove this after having the PR implementing this change, integrated.
```

Comment:
Ugh. If we add new objects to `relatedObjects` to get them must-gathered, then that means we'll end up deleting those objects when you downgrade to a version of CNO where they weren't part of `relatedObjects`. This may be the cause of the "openshift-sdn CRDs get deleted when downgrading" bug? (Not sure where that is or who was working on it...)

We need to avoid doing `deleteRelatedObjectsNotRendered` on downgrade. @squeed 

For now, the fix is probably to backport this hack to 4.6 and 4.5

Reply:
It is the cause of it, @juanluisvaladas verified it with PR: https://github.com/openshift/cluster-network-operator/pull/948. We need this back-ported.